src/bitbox02.js

```diff
@@ -89,6 +89,20 @@ export class BitBox02API {
     constructor(devicePath)  {
         this.devicePath = devicePath;
         this.opened = false;
+        this.socket = null;
+    }
+
+    connectWebsocket() {
+        const socket = new WebSocket("ws://127.0.0.1:8178/api/v1/socket/" + this.devicePath);
+        return new Promise((resolve, reject) => {
+            socket.binaryType = 'arraybuffer'
+            socket.onopen = async function (event) {
```

Comment:
I think the `async` here is not needed, as it uses normal `Promise`

Reply:
inside the function we use `await`, so this function is `async`. didn't work removing the keyword here.

Reply:
sorry i looked at the wrong branch - you are right, fixed